contracts/contracts/coordination/ReimbursementPool.sol

```diff
@@ -0,0 +1,136 @@
+    ///         against malicious operator-miners.
+    uint256 public maxGasPrice;
+
+    event StaticGasUpdated(uint256 newStaticGas);
+
+    event MaxGasPriceUpdated(uint256 newMaxGasPrice);
+
+    event SendingEtherFailed(uint256 refundAmount, address receiver);
+
+    event AuthorizedContract(address thirdPartyContract);
+
+    event UnauthorizedContract(address thirdPartyContract);
+
+    event FundsWithdrawn(uint256 withdrawnAmount, address receiver);
+
+    constructor(uint256 _staticGas, uint256 _maxGasPrice) Ownable(msg.sender) {
+        staticGas = _staticGas;
+        maxGasPrice = _maxGasPrice;
+    }
+
+    /// @notice Receive ETH
+    receive() external payable {}
+
+    /// @notice Refunds ETH to a spender for executing specific transactions.
+    /// @dev Ignoring the result of sending ETH to a receiver is made on purpose.
+    ///      For EOA receiving ETH should always work. If a receiver is a smart
+    ///      contract, then we do not want to fail a transaction, because in some
+    ///      cases the refund is done at the very end of multiple calls where all
+    ///      the previous calls were already paid off. It is a receiver's smart
+    ///      contract resposibility to make sure it can receive ETH.
+    /// @dev Only authorized contracts are allowed calling this function.
+    /// @param gasSpent Gas spent on a transaction that needs to be reimbursed.
+    /// @param receiver Address where the reimbursment is sent.
+    function refund(uint256 gasSpent, address receiver) external nonReentrant {
+        require(isAuthorized[msg.sender], "Contract is not authorized for a refund");
+        require(receiver != address(0), "Receiver's address cannot be zero");
+
+        uint256 gasPrice = tx.gasprice < maxGasPrice ? tx.gasprice : maxGasPrice;
+
+        uint256 refundAmount = (gasSpent + staticGas) * gasPrice;
+
+        /* solhint-disable avoid-low-level-calls */
+        // slither-disable-next-line low-level-calls,unchecked-lowlevel
+        (bool sent, ) = receiver.call{value: refundAmount}("");
+        /* solhint-enable avoid-low-level-calls */
+        if (!sent) {
+            // slither-disable-next-line reentrancy-events
+            emit SendingEtherFailed(refundAmount, receiver);
+        }
+    }
+
+    /// @notice Authorize a contract that can interact with this reimbursment pool.
+    ///         Can be authorized by the owner only.
+    /// @param _contract Authorized contract.
+    function authorize(address _contract) external onlyOwner {
+        isAuthorized[_contract] = true;
+
+        emit AuthorizedContract(_contract);
+    }
+
+    /// @notice Unauthorize a contract that was previously authorized to interact
+    ///         with this reimbursment pool. Can be unauthorized by the
+    ///         owner only.
+    /// @param _contract Authorized contract.
+    function unauthorize(address _contract) external onlyOwner {
```

Comment:
Bit of a nitpick. I know this was from a contract copied from Keep, but should this be `deauthorize`? We've used the `authorize/deauthorize` parlance in other areas.

Same comment for the event.

Reply:
I agree that there's a divergence wrt to staking contracts but this contract is not related to staking, and I'd prefer if we limit changes unless we really need them.

Reply:
Definitely not a "need" 😄